BootloaderCorePkg/Tools/CommonUtility.py

```diff
@@ -211,8 +211,24 @@ def run_process (arg_list, print_cmd = False, capture_out = False):
 
     return output
 
+def get_key_id (priv_key):
+    key_name = os.path.basename(priv_key)
+    # Check for KEY_ID at end of string.
+    if (key_name[-6:] == "KEY_ID"):
+        return key_name
+    else:
+        return None
+
+
 def rsa_sign_file (priv_key, pub_key, hash_type, sign_scheme, in_file, out_file, inc_dat = False, inc_key = False):
 
+    # Check priv_key is path to private key

```

Comment:
We probably do not need this logic here. 
It's better to encapsulate all key logic in single_sign script itself.

Reply:
In container cases specially with layout, key can be passed in either ways,
- $Path\CONTAINER_KEY_ID
- $Path\ContainerTestKey_Priv_RSA2048.pem

In case of $Path\CONTAINER_KEY_ID - we are extracting keyid and pass to single sign script.

I would move this to single sign script to have all handling at one place.